include/aws/s3/private/s3_client_impl.h

```diff
@@ -213,6 +213,13 @@ struct aws_s3_client {
         /* Whether or not the client has started cleaning up all of its resources */
         uint32_t active : 1;
 
+        /* True if the start_destroy function is still executing, which blocks shutdown from completing. */
+        uint32_t start_destroy : 1;
```

Comment:
I feel this name is a little confusing, if I don't know that it's actually the name of the function. So, maybe something like `start_destroy_executing` would be more clear?